servers/services/src/main/java/org/projectnessie/services/rest/TreeResource.java

```diff
@@ -418,19 +450,15 @@ protected Hash doOps(
     }
   }
 
-  private static CommitMeta meta(Principal principal, CommitMeta commitMeta)
-      throws NessieConflictException {
-    if (commitMeta.getCommitter() != null) {
-      throw new NessieConflictException(
-          "Cannot set the committer on the client side. It is set by the server.");
-    }
-    String committer = principal == null ? "" : principal.getName();
-    Instant now = Instant.now();
-    return commitMeta.toBuilder()
+  private CommitMeta meta(final CommitMeta referredCommit, Map<String, String> additionalProps) {
+    final String committer = Optional.ofNullable(getPrincipal()).map(p -> p.getName()).orElse("");
```

Comment:
Let's not add `final` here. It's superfluous.

Reply:
I guess it was a matter of habit that I picked to make use of final keyword liberally, thinking of the advantages and the intent (that a variable not supposed to be reinitialised should be declared final). 

I'll remove it, and try to be consistent with the rest of the code.